core/tests/webdriverio_utils/forms.js

```diff
@@ -251,6 +251,45 @@ var RichTextEditor = async function(elem) {
       }
       await _clickToolbarButton('.cke_button__bulletedlist');
     },
+    // This finds and selects a certain text in the RTE.
+    // NOTE: This only selects the first occurrence of the text.
+    highlightText: async function(text) {
+      // Pre check to ensure that no text is currently highlighted.
+      expect(await browser.execute(() => {
+        return window.getSelection().toString().length;
+      })).toEqual(0);
```

Comment:
Sorry, I see now that my earlier comment wasn't clear. We want this logic in `webdriverio_utils/`, but it needs to be its own function. For example, here's an example from `BlogPages.js`:

https://github.com/oppia/oppia/blob/02c590a0430559cae77593f04b44e2a8d1caa49e/core/tests/webdriverio_utils/BlogPages.js#L121-L126

We need to make this a separate function because someone in the future might write a test that selects some text and then selects some other text without clearing the selection in-between. This is a perfectly reasonable test to write, but putting the pre-check in `highlightText` will cause it to fail.

Instead, I think we want a function in `webdriverio_utils/` that lets us check that nothing is selected. Then we can call that function in `core/tests/webdriverio_desktop/extensions.js`.

This pattern of keeping page interaction logic in the utility files is documented on our [e2e test wiki page](https://github.com/oppia/oppia/wiki/End-to-End-Tests#utilities):

> The webdriverio tests use the above functions to simulate a user interacting with Oppia. They should not interact with the page directly (e.g. using `element()`) but instead make use of the utilities in `webdriverio_utils/`. If new functionality is needed for a test then it should be added in the utilities directory, so that is available for future tests to use and easy to maintain.